core/server/master/src/test/java/alluxio/master/file/FileSystemMasterSyncMetadataMetricsTest.java

```diff
@@ -0,0 +1,506 @@
+import alluxio.util.io.PathUtils;
+
+import com.codahale.metrics.Counter;
+import com.google.common.collect.ImmutableList;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({UnderFileSystem.Factory.class})
+public class FileSystemMasterSyncMetadataMetricsTest {
+  @Rule
+  public TemporaryFolder mTempDir = new TemporaryFolder();
+  private String mUfsUri;
+  private ExceptionThrowingLocalUnderFileSystem mUfs;
+  private ExecutorService mExecutorService;
+  private MasterRegistry mRegistry;
+  private DefaultFileSystemMaster mFileSystemMaster;
+
+  @Before
+  public void before() throws Exception {
+    UserState s = UserState.Factory.create(ServerConfiguration.global());
+    AuthenticatedClientUser.set(s.getUser().getName());
+
+    mTempDir.create();
+    mUfsUri = mTempDir.newFolder().getAbsolutePath();
+
+    mUfs = new ExceptionThrowingLocalUnderFileSystem(new AlluxioURI(mUfsUri),
+        UnderFileSystemConfiguration.defaults(ServerConfiguration.global()));
+    PowerMockito.mockStatic(UnderFileSystem.Factory.class);
+    Mockito.when(UnderFileSystem.Factory.create(anyString(), any())).thenReturn(mUfs);
+
+    ServerConfiguration.set(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.UFS);
+    ServerConfiguration.set(PropertyKey.MASTER_MOUNT_TABLE_ROOT_UFS, mUfsUri);
+    String journalFolderUri = mTempDir.newFolder().getAbsolutePath();
+
+    mExecutorService = Executors
+        .newSingleThreadExecutor(ThreadFactoryUtils.build("DefaultFileSystemMasterTest-%d", true));
+    mRegistry = new MasterRegistry();
+    JournalSystem journalSystem =
+        JournalTestUtils.createJournalSystem(journalFolderUri);
+    CoreMasterContext context = MasterTestUtils.testMasterContext(journalSystem);
+    new MetricsMasterFactory().create(mRegistry, context);
+    BlockMaster blockMaster = new BlockMasterFactory().create(mRegistry, context);
+    mFileSystemMaster = new DefaultFileSystemMaster(blockMaster, context,
+        ExecutorServiceFactories.constantExecutorServiceFactory(mExecutorService));
+    mRegistry.add(FileSystemMaster.class, mFileSystemMaster);
+    journalSystem.start();
+    journalSystem.gainPrimacy();
+    mRegistry.start(true);
+
+    MetricsSystem.resetAllMetrics();
+  }
+
+  @After
+  public void after() throws Exception {
+    mRegistry.stop();
+    mExecutorService.shutdown();
+  }
+
+  @Test
+  public void testMetadataSyncMetrics() throws Exception {
```

Comment:
nit: for unit tests you don't need to name the methods "testXXX" because they are `@Test` already

Reply:
done